src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -95,4 +124,89 @@ export class TestDataProvider implements ITestDataProvider {
   ): boolean {
     return enums.TEST_CODES_FOR_CALCULATING_EXPIRY.CODES.includes(testCode);
   }
+
+  public static applyTestResultsFilters(testResults: models.ITestResult[], filters: models.ITestResultFilters): models.ITestResult[] {
+    const {fromDateTime, toDateTime, testStatus, testStationPNumber, testResultId, testVersion} = filters;
+    testResults = this.filterTestResultsByDeletionFlag(testResults);
+    testResults = this.filterTestTypesByDeletionFlag(testResults);
+    testResults = utils.GetTestResults.filterTestResultByDate(testResults, fromDateTime, toDateTime);
+    if (testStatus) {
+      testResults = this.filterTestResultsByParam(testResults, "testStatus", testStatus);
+    }
+    if (testStationPNumber) {
+      testResults = this.filterTestResultsByParam(testResults, "testStationPNumber", testStationPNumber);
+    }
+    if (testResultId) {
+      testResults = this.filterTestResultsByParam(testResults, "testResultId", testResultId);
+      if (testVersion) {
+        testResults = this.filterTestResultsByTestVersion(testResults, testVersion);
+      }
+    } else {
+      testResults = this.filterTestResultsByTestVersion(testResults, enums.TEST_VERSION.CURRENT);
+      testResults = this.removeTestHistory(testResults);
+    }
+
+    return testResults;
+  }
+
+  public static filterTestResultsByParam(testResults: models.ITestResult[], filterName: string, filterValue: any): models.ITestResult[] {
+    return testResults.filter((testResult) => {
+      const k = filterName as keyof typeof testResult;
+      return testResult[k] === filterValue;
+    });
+  }
+
+  public static filterTestResultByDate(testResults: models.ITestResult[], fromDateTime: string | number | Date, toDateTime: string | number | Date): models.ITestResult[] {
+
+    return testResults.filter((testResult: { testStartTimestamp: string | number | Date; testEndTimestamp: string | number | Date; }) => {
+      return moment(testResult.testStartTimestamp).isAfter(fromDateTime) && moment(testResult.testEndTimestamp).isBefore(toDateTime);
```

Comment:
Can we push this into the date utils so that this class doesn't need to use moment.  Keeps it cleaner.

Reply:
done